rpc/src/v1/impls/signing.rs

```diff
@@ -125,11 +125,24 @@ impl<D: Dispatcher + 'static> SigningQueueClient<D> {
 						.map(dispatch::WithToken::into_value)
 						.map(DispatchResult::Value))
 				} else {
-					Either::B(future::done(
-						signer.add_request(payload, origin)
-							.map(|(id, future)| DispatchResult::Future(id, future))
-							.map_err(|_| errors::request_rejected_limit())
-					))
+					if signer.is_enabled() {
+						Either::B(future::done(
+							signer.add_request(payload, origin)
+								.map(|(id, future)| {
+									DispatchResult::Future(id, future)
+								})
+								.map_err(|_| errors::request_rejected_limit())
+						))
+					} else {
+						Either::B(
+							future::done(
+								Err(Error {
+									code: ErrorCode::ServerError(errors::codes::UNSUPPORTED_REQUEST),
```

Comment:
we have this error https://github.com/paritytech/parity-ethereum/blob/be9d4d9faa8400799064200bcbdc82dc827b29d1/rpc/src/v1/helpers/errors.rs#L383-L390
or is it different?

Reply:
oh cool, didn't see this